In [33]:
import numpy as np
import pandas as pd

#raw data as a csv file
filepath = 'NBASTATS2023.csv' 

#display all rows/columns
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

#read csv into specific dataframe
df = pd.read_csv(filepath)

#shows how many players and fields we are working with

print(df.shape) 

#uncomment below to see full dataset
#print(df)

#check for missing values in our dataframe
#ensure dataset was formatted and parsed ok

missing_values = df.isnull().sum()

print(missing_values)

#retrieve values for a specific player
print("Bam Adebayo Stats \n")
df.loc[df['NAME'] == 'Bam Adebayo']




(609, 29)
RANK     609
NAME       0
TEAM       0
POS        0
AGE        0
GP         0
MPG        0
USG%       0
TO%        1
FTA        0
FT%        0
2PA        0
2P%        0
3PA        0
3P%        0
eFG%       3
TS%        3
PPG        0
RPG        0
APG        0
SPG        0
BPG        0
TPG        0
P+R        0
P+A        0
P+R+A      0
VI         0
ORtg      24
DRtg      23
dtype: int64
Bam Adebayo Stats 



RANK         NAME TEAM  POS   AGE  GP   MPG  USG%   TO%  FTA    FT%   2PA  \
2   NaN  Bam Adebayo  Mia  C-F  25.7  75  34.6  25.3  14.4  402  0.806  1102   

     2P%  3PA    3P%   eFG%    TS%   PPG  RPG  APG  SPG  BPG  TPG   P+R   P+A  \
2  0.545   12  0.083  0.541  0.592  20.4  9.2  3.2  1.2  0.8  2.5  29.6  23.6   

   P+R+A   VI   ORtg   DRtg  
2   32.8  9.8  116.0  105.6

CLEANING

We can see that rank shows as invalid(NaN) for all players(if we uncomment line 19) , so lets drop rank from our dataset

We also can do further cleaning of our dataset. In this case, players whos stats cannot be calculated for some fields(NaN) likely have played not enough games, or have not gotten >0 in a certain stat. We can trim our dataset by either filtering players out who have played very few minutes/games(option 1), or we can just remove any player that has a null value for any stat(option 2), or input filler numbers for our NaN values(option 3). If you dont know exactly how some of the advanced stats are calculated, option 3 is risky, so we will focus on the first two options

In [5]:
df = df.drop('RANK', axis=1)
#show new shape, should have 1 less column than previously
df.shape

(609, 28)

In [6]:


#Option 1
#removes players who ahve played less than 10 games and average less than 
#minutes per game
option_1_df = df.loc[(df['GP']>10) & (df['MPG']>5)]
print(option_1_df.shape)

missing_values = option_1_df.isnull().sum()

print(missing_values)



(503, 28)
NAME     0
TEAM     0
POS      0
AGE      0
GP       0
MPG      0
USG%     0
TO%      0
FTA      0
FT%      0
2PA      0
2P%      0
3PA      0
3P%      0
eFG%     0
TS%      0
PPG      0
RPG      0
APG      0
SPG      0
BPG      0
TPG      0
P+R      0
P+A      0
P+R+A    0
VI       0
ORtg     0
DRtg     0
dtype: int64


We can see that we have trimmed 106 players from our list using the minutes/games restriction, and have in fact gotten
rid of all the players with missing values. However, this is likely overkill as some players have played less than 10 games and less than 5 minutes per game while having analyzable values.


In [8]:
#Option 2
#remove only players who have an invalid value
o2_df= df.loc[(df['ORtg'].notnull()) & (df['eFG%'].notnull())]
print(o2_df.shape)
#we can add in all the other or clauses, but these 2 do the trick due to how these stats overlap

missing_values = o2_df.isnull().sum()

print(missing_values)

(585, 28)
NAME     0
TEAM     0
POS      0
AGE      0
GP       0
MPG      0
USG%     0
TO%      0
FTA      0
FT%      0
2PA      0
2P%      0
3PA      0
3P%      0
eFG%     0
TS%      0
PPG      0
RPG      0
APG      0
SPG      0
BPG      0
TPG      0
P+R      0
P+A      0
P+R+A    0
VI       0
ORtg     0
DRtg     0
dtype: int64


Now we can see that we have elimnated only 24 players and still have all analyzable values

In [10]:
#I would like to see players who score over 20 points per game
#lets store this as a new dataframe and we can do some transformations
pts_over_20 = o2_df.loc[o2_df['PPG'] >= 20]
print(pts_over_20.shape)

#From this we can see that there are 60 players in the 2022-23 NBA season who averaged over 20 points each game



(60, 28)


To do some more advanced filtering: 
I want to separate elite shooters(1) from strong defensive
players(2), from those who are neither(3), or both(4). We are going to create a new field to tag our players with, and set some conditions for each category. 

In order to qualify as an elite shooter:
- over 50 games played(large sample size,82 games is max)
- score over 20 points per game(ppg)
- High efficiency(Free throw% over 80%, Field goal percentage over 35%, True shooting % over 55%)

In order to qualify as a strong defensive player:
- over 50 games played
- 107 or lower defensive rating(league average is 110, lower is better)
- 1 block or steal per game




In [12]:
elite_shooters = pts_over_20.loc[(pts_over_20['GP']>50) & (pts_over_20['TS%']> .55) 
                                 & (pts_over_20['eFG%'] >.35) & (pts_over_20['FT%']>.80)]
print(elite_shooters.shape)
elite_shooters

(31, 28)


NAME TEAM  POS   AGE  GP   MPG  USG%   TO%  FTA  \
2                Bam Adebayo  Mia  C-F  25.7  75  34.6  25.3  14.4  402   
26              Desmond Bane  Mem    G  24.8  58  31.8  26.1  11.8  205   
53          Bojan Bogdanovic  Det    F  34.0  59  32.1  25.9  13.3  303   
57              Devin Booker  Pho    G  26.4  53  34.6  31.8  12.0  358   
77             Jalen Brunson  Nyk    G  26.6  68  35.0  27.2  10.6  392   
85              Jimmy Butler  Mia    F  33.6  64  33.4  25.6  10.2  555   
102          Jordan Clarkson  Uta    G  30.8  61  32.6  27.8  15.3  245   
116            Stephen Curry  Gol    G  35.1  56  34.7  31.0  13.6  281   
125            DeMar DeRozan  Chi  G-F  33.7  74  36.2  27.8  10.5  523   
154              Joel Embiid  Phi  C-F  29.1  66  34.6  37.0  14.5  771   
170           Darius Garland  Cle    G  23.2  69  35.5  26.9  15.0  322   
174              Paul George  Lac    F  32.9  56  34.6  29.5  14.9  294   
177  Shai Gilgeous-Alexander  Okc  G-F  24.7  68  35.5  32.8  12.2  739   
186             Jerami Grant  Por    F  29.1  63  35.6  22.8  11.3  342   
201        Tyrese Haliburton  Ind    G  23.1  56  33.6  23.9  14.4  203   
206             James Harden  Phi    G  33.6  58  36.8  25.0  18.8  361   
224              Tyler Herro  Mia    G  23.2  67  34.9  25.6  12.5  182   
270             Nikola Jokic  Den    C  28.1  69  33.7  27.2  19.5  415   
307              Zach LaVine  Chi  G-F  28.1  77  36.0  28.3  12.3  428   
314            Kawhi Leonard  Lac    F  31.8  52  33.6  27.0   9.2  279   
318           Damian Lillard  Por    G  32.7  58  36.3  33.8  13.7  558   
335          Lauri Markkanen  Uta  F-C  25.9  66  34.4  26.6  10.0  399   
345             Tyrese Maxey  Phi    G  22.4  60  33.6  24.1   8.1  213   
365         Donovan Mitchell  Cle    G  26.6  68  35.8  32.1  11.4  368   
379             Jamal Murray  Den    G  26.1  65  32.8  26.1  12.2  216   
422             Jordan Poole  Gol    G  23.8  82  30.0  29.2  16.5  415   
427       Kristaps Porzingis  Was  F-C  27.7  65  32.6  27.4  11.8  416   
483          Anfernee Simons  Por    G  23.8  62  35.0  25.1  10.9  179   
501             Jayson Tatum  Bos  F-G  25.1  74  36.9  32.7  12.0  622   
508            Klay Thompson  Gol    G  33.2  69  33.0  26.4   8.9  132   
574               Trae Young  Atl    G  24.6  73  34.8  32.6  17.7  639   

       FT%   2PA    2P%  3PA    3P%   eFG%    TS%   PPG   RPG   APG  SPG  BPG  \
2    0.806  1102  0.545   12  0.083  0.541  0.592  20.4   9.2   3.2  1.2  0.8   
26   0.883   532  0.534  407  0.408  0.568  0.606  21.5   5.0   4.4  1.0  0.4   
53   0.884   529  0.539  353  0.411  0.570  0.627  21.6   3.8   2.6  0.6  0.1   
57   0.855   751  0.554  316  0.351  0.546  0.601  27.8   4.5   5.5  1.0  0.3   
77   0.829   873  0.519  322  0.416  0.547  0.597  24.0   3.5   6.2  0.9  0.2   
85   0.850   785  0.564  103  0.350  0.560  0.647  22.9   5.8   5.3  1.8  0.3   
102  0.816   572  0.530  459  0.338  0.519  0.558  20.8   4.0   4.4  0.5  0.2   
116  0.915   494  0.579  639  0.427  0.614  0.656  29.4   6.1   6.3  0.9  0.4   
125  0.872  1161  0.526  142  0.324  0.522  0.592  24.5   4.6   5.1  1.1  0.5   
154  0.857  1128  0.587  200  0.330  0.573  0.655  33.1  10.2   4.2  1.0  1.7   
170  0.863   717  0.492  412  0.410  0.537  0.587  21.6   2.7   7.8  1.2  0.1   
174  0.871   578  0.521  426  0.371  0.536  0.588  23.8   6.1   5.1  1.5  0.4   
177  0.905  1213  0.533  168  0.345  0.531  0.626  31.4   4.8   5.5  1.6  1.0   
186  0.813   554  0.523  359  0.401  0.554  0.606  20.5   4.5   2.4  0.8  0.8   
201  0.867   439  0.572  402  0.400  0.586  0.624  20.7   3.7  10.4  1.6  0.4   
206  0.867   424  0.495  419  0.384  0.536  0.607  21.0   6.1  10.7  1.2  0.5   
224  0.934   572  0.497  537  0.378  0.531  0.566  20.1   5.4   4.2  0.8  0.2   
270  0.822   873  0.675  149  0.383  0.660  0.701  24.5  11.8   9.8  1.3  0.7   
307  0.848   844  0.556  544  0.375  0.558  0.607  24.8   4.5   4.2  0.9  0.2   
314

We can see we have a pool of 32 players who meet our criteria for elite shooters

In [14]:
strong_defense = o2_df.loc[o2_df['GP']>20]
strong_defense = strong_defense.loc[(strong_defense['DRtg']<107) 
                                    & ((strong_defense['SPG']>=1)|(strong_defense['BPG']>=1))]
print(strong_defense.shape)
strong_defense

(47, 28)


NAME TEAM  POS   AGE  GP   MPG  USG%   TO%  FTA    FT%  \
1           Steven Adams  Mem    C  29.7  42  27.0  14.6  23.1  129  0.364   
2            Bam Adebayo  Mia  C-F  25.7  75  34.6  25.3  14.4  402  0.806   
8          Jarrett Allen  Cle    C  25.0  68  32.6  16.5  12.9  221  0.733   
9          Jose Alvarado  Nor    G  25.0  61  21.5  19.8  14.2   80  0.813   
23              Mo Bamba  Orl    C  24.9  40  17.0  16.3   9.3   51  0.686   
45      Patrick Beverley  Chi    G  34.7  22  27.5  10.5  13.1   15  0.533   
51       Bismack Biyombo  Pho    C  30.6  61  14.3  14.0  19.8   70  0.357   
73          Jaylen Brown  Bos  G-F  26.5  67  35.9  31.5  12.5  344  0.765   
94           Alex Caruso  Chi    G  29.1  67  23.5  11.1  21.2   73  0.808   
103          Nic Claxton  Bro  F-C  24.0  76  29.9  15.5  14.1  246  0.541   
118        Anthony Davis  Lal  F-C  30.1  56  34.0  28.4  11.2  444  0.784   
154          Joel Embiid  Phi  C-F  29.1  66  34.6  37.0  14.5  771  0.857   
156       Bruno Fernando  Hou  F-C  24.7  31  11.7  15.1  17.7   44  0.682   
169       Daniel Gafford  Was  F-C  24.5  78  20.6  15.2  17.7  187  0.679   
179          Rudy Gobert  Min    C  30.8  70  30.7  16.3  18.3  340  0.644   
190       Draymond Green  Gol    F  33.1  73  31.5  13.2  30.1  108  0.713   
216            Josh Hart  Nyk    G  28.1  25  30.0  12.6  19.5   57  0.789   
252    Jaren Jackson Jr.  Mem  F-C  23.6  63  28.4  24.7  11.5  306  0.788   
254       Isaiah Jackson  Ind    F  21.2  63  16.5  17.5  14.0  146  0.651   
270         Nikola Jokic  Den    C  28.1  69  33.7  27.2  19.5  415  0.822   
275        Herbert Jones  Nor    F  24.5  66  29.5  14.5  14.9  161  0.764   
296     Christian Koloko  Tor    C  22.8  58  13.8  10.4  11.7   59  0.627   
297         John Konchar  Mem    G  27.0  72  20.7  10.4  10.4   36  0.778   
316         Caris LeVert  Cle    G  28.6  74  30.3  18.7  13.5  180  0.722   
323          Brook Lopez  Mil    C  35.0  78  30.4  19.3  10.6  190  0.784   
350      Jalen McDaniels  Cha  F-C  25.2  56  26.7  17.4  13.8  104  0.846   
357    De'Anthony Melton  Phi    G  24.9  77  27.9  16.5  13.5   87  0.793   
366          Evan Mobley  Cle    C  21.8  79  34.4  20.2  13.3  303  0.673   
402       Onyeka Okongwu  Atl  F-C  22.3  80  23.1  15.1  14.0  196  0.781   
405       Victor Oladipo  Mia    G  30.9  42  26.3  21.1  17.7   79  0.747   
420         Jakob Poeltl  Tor    C  27.5  26  27.2  17.5  11.2   72  0.569   
447        Nick Richards  Cha    C  25.4  65  18.7  15.7  15.8  183  0.749   
449      Josh Richardson  Nor    G  29.6  23  23.2  14.5  13.4   21  0.762   
453    Mitchell Robinson  Nyk  C-F  25.0  59  26.9  10.1  12.8  122  0.484   
481          Ben Simmons  Bro  G-F  26.7  42  26.3  14.3  29.2   57  0.439   
485         Marcus Smart  Bos    G  29.1  61  32.1  17.8  19.2  118  0.746   
497          Jalen Suggs  Orl    G  21.9  53  23.5  20.4  17.5  119  0.723   
501         Jayson Tatum  Bos  F-G  25.1  74  36.9  32.7  12.0  622  0.854   
512       Xavier Tillman  Mem    F  24.2  61  19.3  13.7  12.6   89  0.551   
523    Jarred Vanderbilt  Uta    F  24.0  52  24.1  14.1  17.9  102  0.657   
524    Jarred Vanderbilt  Lal    F  24.0  26  24.1  12.5  17.1   37  0.784   
548    Russell Westbrook  Lal    G  34.4  52  28.7  28.9  20.2  238  0.655   
556  Robert Williams III  Bos  C-F  25.5  35  23.5  11.8  16.6   41  0.610   
563        Mark Williams  Cha    C  21.3  43  19.3  17.0  13.7   97  0.691   
570       Christian Wood  Dal    F  27.5  67  25.9  26.4  13.6  281  0.772   
573       Thaddeus Young  Tor    F  34.8  54  14.7  13.5  17.5   26  0.692   
576          Ivica Zubac  Lac    C  26.1  76  28.5  14.8  18.5  238  0.697   

      2PA    2P%  3PA    3P%   eFG%    TS%   PPG   RPG  APG  SPG  BPG  TPG  \
1     262  0.599    1  0.000  0.597  0.564   8.6  11.5  2.3  0.9  1.1  1.9   
2    1102  0.545   12  0.083  0.541  0.592  20.4   9.2  3.2  1.2  0.8  2.5   
8     616  0.653   10  0.100  0.645  0.67

We have 47 players who meet our criteria as strong defenders

Next lets move our criteria to value for fantasy games. Fantasy basketball for the league I am in uses the following modifiers: 

Points Scored:1 Points

Total Rebounds:1.2 Points

Assists:1.5 Points

Steals:3 Points

Blocked Shots:3 Points

Turnovers:-1 Points

In [17]:
#Lets create a fantasy points category and append it to our dataframe, maxing out at two decimal points

from math import trunc

#truncate our values so that its rounded to 2 decimals
def truncate(num):
    num = trunc(num*100)/100
    return num
fp = []

for index, row in o2_df.iterrows():
    fanpoints = truncate(row['PPG'] + (1.2 * row['RPG']) + (1.5 * row['APG']) + (3*(row['SPG'] +row['BPG'])) + row['TPG'])
    fp.append(fanpoints)


  


In [101]:
o2_df['FANPOINTS'] = fp
o2_df.iloc[0]
o2_df[['NAME','FANPOINTS']].sort_values(by='FANPOINTS', ascending = False)

NAME  FANPOINTS
134               Luka Doncic      64.02
154               Joel Embiid      63.14
270              Nikola Jokic      62.96
11      Giannis Antetokounmpo      62.51
259              LeBron James      56.76
118             Anthony Davis      56.42
146              Kevin Durant      56.09
177   Shai Gilgeous-Alexander      56.00
501              Jayson Tatum      55.86
318            Damian Lillard      55.80
116             Stephen Curry      53.27
372                 Ja Morant      53.03
206              James Harden      52.87
574                Trae Young      52.80
22                LaMelo Ball      51.98
467          Domantas Sabonis      51.61
564           Zion Williamson      49.80
249              Kyrie Irving      49.60
201         Tyrese Haliburton      49.24
248              Kyrie Irving      48.87
479             Pascal Siakam      48.86
438             Julius Randle      48.75
365          Donovan Mitchell      48.36
150           Anthony Edwards      48.36
57               Devin Booker      48.05
174               Paul George      47.57
73               Jaylen Brown      47.53
427        Kristaps Porzingis      46.63
247            Brandon Ingram      46.59
147              Kevin Durant      46.23
166              De'Aaron Fox      45.89
85               Jimmy Butler      45.71
314             Kawhi Leonard      44.85
2                 Bam Adebayo      44.73
517        Karl-Anthony Towns      44.62
125             DeMar DeRozan      44.57
606              RaiQuan Gray      44.30
335           Lauri Markkanen      44.27
525             Fred VanVleet      44.22
231              Jrue Holiday      44.22
114           Cade Cunningham      43.83
39               Bradley Beal      43.68
378           Dejounte Murray      43.61
170            Darius Garland      43.34
77              Jalen Brunson      42.90
423          Kevin Porter Jr.      42.41
548         Russell Westbrook      42.39
307               Zach LaVine      42.30
66              Mikal Bridges      42.14
252         Jaren Jackson Jr.      41.96
176               Josh Giddey      41.78
302                Kyle Kuzma      41.69
529            Nikola Vucevic      41.50
132         Spencer Dinwiddie      41.37
347               CJ McCollum      41.33
549         Russell Westbrook      41.27
25             Paolo Banchero      40.52
26               Desmond Bane      40.50
461              Terry Rozier      40.27
412                Chris Paul      40.01
379              Jamal Murray      39.90
366               Evan Mobley      39.90
520              Myles Turner      39.50
474            Alperen Sengun      39.45
18              Deandre Ayton      38.54
224               Tyler Herro      38.28
463          D'Angelo Russell      38.12
30             Scottie Barnes      38.11
103               Nic Claxton      37.98
192               Jalen Green      37.69
179               Rudy Gobert      37.42
102           Jordan Clarkson      37.30
267            Keldon Johnson      37.15
595               Skylar Mays      36.89
422              Jordan Poole      36.79
323               Brook Lopez      36.29
186              Jerami Grant      36.09
8               Jarrett Allen      36.01
420              Jakob Poeltl      35.92
397              Jusuf Nurkic      35.67
464          D'Angelo Russell      35.62
14               O.G. Anunoby      35.60
190            Draymond Green      35.54
508             Klay Thompson      35.51
419              Jakob Poeltl      35.35
167            Markelle Fultz      35.22
483           Anfernee Simons      35.17
65              Mikal Bridges      35.15
251                Jaden Ivey      34.98
181              Aaron Gordon      34.91
411           Kelly Oubre Jr.      34.88
526             Devin Vassell      34.58
531              Franz Wagner      34.47
53           Bojan Bogdanovic      34.46
570            Christian Wood      34.36
417             Mason Plumlee      34.36
92         Wendell Carter Jr.      34.29
522    

In [316]:
import pandas as pd

#raw data as a csv file
filepath = 'NBASTATS2023.csv' 

#display all rows/columns
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
rankings = pd.read_csv('FantasyPros_2023_Overall_NBA_ADP_Rankings.csv')

In [318]:
print(rankings.shape)
rankings.iloc[0]


(258, 8)


Rank                  1.0
Player       Nikola Jokic
Team                  DEN
Positions           C DTD
Yahoo                 1.0
ESPN                  1.0
CBS                   1.0
AVG                   1.0
Name: 0, dtype: object

This is a table that contains the average pick each player is taken in fantasy drafts across different websites.

In fantasy basketball, higher average draft pick means that the player has higher value when you are picking(opponent may select). Lets combine our rankings and stats dataframes and create a new column for weighted fantasy value(Fantasy points per game /(1.5* AVG draft pick)

In [324]:
weighted = o2_df.copy(deep = True)
print(weighted.shape)
weightedFP = []
rankings = rankings.rename(columns = {'Player':'NAME'})
#print(rankings.head)
result = rankings.merge(weighted, how = 'inner', on = 'NAME')
print(result.shape)
#208 overlapping players
#print(result.iloc[0])


for index,row in result.iterrows():
    wfpg = row['FANPOINTS']/(1.5*row['AVG'])
    weightedFP.append(wfpg)
    
result['WFPG'] = weightedFP
result.shape   
result


(585, 29)
(239, 36)


Rank                      NAME Team      Positions  Yahoo   ESPN    CBS  \
0      1.0              Nikola Jokic  DEN          C DTD    1.0    1.0    1.0   
1      2.0               Luka Doncic  DAL             PG    2.0    3.0    3.0   
2      3.0     Giannis Antetokounmpo  MIL           PF,C    6.0    2.0    2.0   
3      4.0               Joel Embiid  PHI              C    3.0    5.0    4.0   
4      5.0              Jayson Tatum  BOS          SF,PF    4.0    4.0    6.0   
5      6.0   Shai Gilgeous-Alexander  OKC          PG,SG    5.0    6.0    5.0   
6      7.0             Stephen Curry  GSW          PG,SG    7.0    8.0    9.0   
7      8.0         Tyrese Haliburton  IND          PG,SG    8.0    7.0   10.0   
8      9.0            Damian Lillard  MIL             PG   10.0   11.0    7.0   
9     10.0              Kevin Durant  PHO          SF,PF    9.0   15.0    8.0   
10    10.0              Kevin Durant  PHO          SF,PF    9.0   15.0    8.0   
11    11.0          Domantas Sabonis  SAC           PF,C   15.0   10.0   11.0   
12    12.0             Anthony Davis  LAL       PF,C DTD   12.0   17.0   13.0   
13    13.0           Anthony Edwards  MIN          SG,SF   14.0    9.0   20.0   
14    14.0                Trae Young  ATL             PG   17.0   14.0   12.0   
15    15.0               LaMelo Ball  CHA          PG,SG   11.0   19.0   16.0   
16    16.0              Devin Booker  PHO      SF,SG OUT   16.0   13.0   18.0   
17    17.0              LeBron James  LAL          PF,SF   23.0   12.0   14.0   
18    18.0          Donovan Mitchell  CLE      PG,SG DTD   18.0   18.0   21.0   
19    20.0              Kyrie Irving  DAL          PG,SG   13.0   24.0   27.0   
20    20.0              Kyrie Irving  DAL          PG,SG   13.0   24.0   27.0   
21    21.0             Mikal Bridges  BKN          SG,SF   19.0   16.0   30.0   
22    21.0             Mikal Bridges  BKN          SG,SF   19.0   16.0   30.0   
23    22.0             Pascal Siakam  TOR           PF,C   29.0   26.0   17.0   
24    23.0              Jimmy Butler  MIA      PF,SF DTD   24.0   29.0   22.0   
25    24.0        Karl-Anthony Towns  MIN           C,PF   21.0   36.0   23.0   
26    25.0              De'Aaron Fox  SAC         PG OUT   38.0   25.0   24.0   
27    26.0               Bam Adebayo  MIA          C DTD   32.0   27.0   29.0   
28    27.0           Lauri Markkanen  UTH          PF,SF   26.0   23.0   43.0   
29    28.0           Dejounte Murray  ATL          PG,SG   33.0   31.0   33.0   
30    29.0             Jalen Brunson  NYK          PG,SG   36.0   21.0   42.0   
31    30.0         Jaren Jackson Jr.  MEM           PF,C   20.0   34.0   46.0   
32    31.0            Nikola Vucevic  CHI              C   44.0   28.0   28.0   
33    32.0              Jaylen Brown  BOS          SG,SF   35.0   35.0   31.0   
34    33.0             DeMar DeRozan  CHI      SF,SG DTD   45.0   30.0   32.0   
35    34.0             Kawhi Leonard  LAC       PF,SF,SG   27.0   40.0   41.0   
36    35.0             Julius Randle  NYK             PF   54.0   38.0   19.0   
37    36.0            Darius Garland  CLE         PG DTD   39.0   39.0   34.0   
38    37.0           Zion Williamson  NOR       C,PF OUT   55.0   33.0   25.0   
39    38.0              James Harden  LAC          PG,SG   30.0   45.0   38.0   
40    39.0        Kristaps Porzingis  BOS       PF,C DTD   43.0   37.0   35.0   
41    40.0              Jordan Poole  WAS          PG,SG   46.0   22.0   48.0   
42    41.0            Brandon Ingram  NOR   PF,SF,SG OUT   53.0   42.0   26.0   
43    42.0              Desmond Bane  MEM          SF,SG   22.0   46.0   55.0   
44    43.0           Cade Cunningham  DET          PG,SG   34.0   43.0   47.0   
45    44.0              Jamal Murray  DEN      PG,SG OUT   40.0   44.0   40.0   
46    45.0             Fred VanVleet  HOU             PG   28.0   59.0   39.0   
47    46.0               Paul George  LAC       SF,PF,SG   31.0   50.0   45.0   
48    47.0            Paolo Banchero 

Our rankings data contains 258 players, and not all of them match the data in our previous dataframe so in this case where our data is mismatched size, we will compute weighted fan point values only for those players that are contained in both datasets(239). 

In [328]:
result.sort_values('WFPG', ascending = False)

Rank                      NAME Team      Positions  Yahoo   ESPN    CBS  \
0      1.0              Nikola Jokic  DEN          C DTD    1.0    1.0    1.0   
1      2.0               Luka Doncic  DAL             PG    2.0    3.0    3.0   
2      3.0     Giannis Antetokounmpo  MIL           PF,C    6.0    2.0    2.0   
3      4.0               Joel Embiid  PHI              C    3.0    5.0    4.0   
4      5.0              Jayson Tatum  BOS          SF,PF    4.0    4.0    6.0   
5      6.0   Shai Gilgeous-Alexander  OKC          PG,SG    5.0    6.0    5.0   
6      7.0             Stephen Curry  GSW          PG,SG    7.0    8.0    9.0   
8      9.0            Damian Lillard  MIL             PG   10.0   11.0    7.0   
7      8.0         Tyrese Haliburton  IND          PG,SG    8.0    7.0   10.0   
9     10.0              Kevin Durant  PHO          SF,PF    9.0   15.0    8.0   
10    10.0              Kevin Durant  PHO          SF,PF    9.0   15.0    8.0   
11    11.0          Domantas Sabonis  SAC           PF,C   15.0   10.0   11.0   
12    12.0             Anthony Davis  LAL       PF,C DTD   12.0   17.0   13.0   
14    14.0                Trae Young  ATL             PG   17.0   14.0   12.0   
17    17.0              LeBron James  LAL          PF,SF   23.0   12.0   14.0   
15    15.0               LaMelo Ball  CHA          PG,SG   11.0   19.0   16.0   
13    13.0           Anthony Edwards  MIN          SG,SF   14.0    9.0   20.0   
16    16.0              Devin Booker  PHO      SF,SG OUT   16.0   13.0   18.0   
18    18.0          Donovan Mitchell  CLE      PG,SG DTD   18.0   18.0   21.0   
20    20.0              Kyrie Irving  DAL          PG,SG   13.0   24.0   27.0   
19    20.0              Kyrie Irving  DAL          PG,SG   13.0   24.0   27.0   
23    22.0             Pascal Siakam  TOR           PF,C   29.0   26.0   17.0   
22    21.0             Mikal Bridges  BKN          SG,SF   19.0   16.0   30.0   
24    23.0              Jimmy Butler  MIA      PF,SF DTD   24.0   29.0   22.0   
25    24.0        Karl-Anthony Towns  MIN           C,PF   21.0   36.0   23.0   
21    21.0             Mikal Bridges  BKN          SG,SF   19.0   16.0   30.0   
26    25.0              De'Aaron Fox  SAC         PG OUT   38.0   25.0   24.0   
27    26.0               Bam Adebayo  MIA          C DTD   32.0   27.0   29.0   
28    27.0           Lauri Markkanen  UTH          PF,SF   26.0   23.0   43.0   
33    32.0              Jaylen Brown  BOS          SG,SF   35.0   35.0   31.0   
39    38.0              James Harden  LAC          PG,SG   30.0   45.0   38.0   
29    28.0           Dejounte Murray  ATL          PG,SG   33.0   31.0   33.0   
38    37.0           Zion Williamson  NOR       C,PF OUT   55.0   33.0   25.0   
36    35.0             Julius Randle  NYK             PF   54.0   38.0   19.0   
30    29.0             Jalen Brunson  NYK          PG,SG   36.0   21.0   42.0   
31    30.0         Jaren Jackson Jr.  MEM           PF,C   20.0   34.0   46.0   
34    33.0             DeMar DeRozan  CHI      SF,SG DTD   45.0   30.0   32.0   
32    31.0            Nikola Vucevic  CHI              C   44.0   28.0   28.0   
35    34.0             Kawhi Leonard  LAC       PF,SF,SG   27.0   40.0   41.0   
40    39.0        Kristaps Porzingis  BOS       PF,C DTD   43.0   37.0   35.0   
37    36.0            Darius Garland  CLE         PG DTD   39.0   39.0   34.0   
42    41.0            Brandon Ingram  NOR   PF,SF,SG OUT   53.0   42.0   26.0   
47    46.0               Paul George  LAC       SF,PF,SG   31.0   50.0   45.0   
44    43.0           Cade Cunningham  DET          PG,SG   34.0   43.0   47.0   
46    45.0             Fred VanVleet  HOU             PG   28.0   59.0   39.0   
43    42.0              Desmond Bane  MEM          SF,SG   22.0   46.0   55.0   
45    44.0              Jamal Murray  DEN      PG,SG OUT   40.0   44.0   40.0   
41    40.0              Jordan Poole  WAS          PG,SG   46.0   22.0   48.0   
48    47.0            Paolo Banchero 

We have now merged our dataframes, 